doc/manual/spec

```diff
@@ -214,24 +214,67 @@ can express this as
 	BuildConflicts: gcc <= 2.7.2.1
 \endverbatim
 
+\section comments Comments
+
+Comments in spec file have # at the start of the line.
+\verbatim
+	# this is a comment
+\endverbatim
+Macros are expanded in comments, thus if a macro is in comment use two percent
+characters before it:
+\verbatim
+	# make unversioned %%__python an error unless explicitly overridden
+\endverbatim
+
 \section conditionals Conditionals
 
-RPM's spec file format allows conditional blocks of code to be used dependent
-on various properties such as architecture (%ifarch / %ifnarch), operating system
-(%ifos / %ifnos), or any other conditional expression (%if). Conditionals may be
-nested within other conditionals:
+RPM's spec file format allows conditional blocks of code to be used
+depending on various properties such as architecture (%ifarch /%ifnarch),
+operating system (%ifos / %ifnos), or a conditional expression (%if).
 
+%ifarch is generally used for building RPM packages for multiple
+platforms like:
 \verbatim
-	%ifarch ppc64 ppc x86_64
-	...
-	%if 0%{?ver} && 0%{?ver} >= 100
-	...
-	%else
-	...
-	%endif
+	%ifarch %{power64} s390x x86_64 ia64
+	%global architectures 1
 	%endif
 \endverbatim
 
-Conditionals are not macros, thus they can be used in spec files only.
+%ifos is used to control RPM's spec file processing according to the
+build system's operating system.
```

Comment:
It's not build system's operating system, but the build *target* operating system. In > 99% of the cases that's the same thing, but in theory rpm supports cross-architecture and -os builds.